tests/test_handle_exceptions.py

```diff
@@ -0,0 +1,129 @@
+        self.assertIn('handle_exceptions got error for function', str(cm.output))
+
+        with self.assertLogs() as cm:
+            resp_json = self.get_data()
+            self.assertEqual(resp_json.status_code, 500)
+            resp_msg_data = json.loads(resp_json.content.decode('utf-8'))
+            self.assertIsInstance(resp_msg_data, dict)
+            self.assertEqual(resp_msg_data['error'], 'Internal Server Error')
+        self.assertIn('handle_exceptions got error for function', str(cm.output))
+
+        with self.assertLogs() as cm:
+            resp_json = self.raise_assertion()
+            self.assertEqual(resp_json.status_code, 500)
+            resp_json_data = json.loads(resp_json.content.decode('utf-8'))
+            self.assertIsInstance(resp_json_data, dict)
+            self.assertEqual(resp_json_data['error'], 'Internal Server Error')
+        self.assertIn('AssertionError', str(cm.output))
+        self.assertIn('handle_exceptions got error for function', str(cm.output))
+
+    def test_raises_exception_flask(self):
+        web_util = WebHelpersUtils()
+        web_util.init(web_util.ServerType.FLASK)
+
+        with self.assertLogs() as cm:
+            resp_json = self.raise_exception(BaseException)
+            self.assertEqual(resp_json.status_code, 500)
+            self.assertEqual(resp_json.status, "500 INTERNAL SERVER ERROR")
+            resp_msg_data = json.loads(resp_json.data.decode('utf-8'))
+            self.assertIsInstance(resp_msg_data, dict)
+            self.assertEqual(resp_msg_data['error'], 'Internal Server Error')
+        self.assertIn('BaseException', str(cm.output))
+        self.assertIn('handle_exceptions got error for function', str(cm.output))
+
+        with self.assertLogs() as cm:
+            resp_json = self.raise_exception(AssertionError)
+            self.assertEqual(resp_json.status_code, 500)
+            self.assertEqual(resp_json.status, "500 INTERNAL SERVER ERROR")
+            resp_msg_data = json.loads(resp_json.data.decode('utf-8'))
+            self.assertIsInstance(resp_msg_data, dict)
+            self.assertEqual(resp_msg_data['error'], 'Internal Server Error')
+        self.assertIn('AssertionError', str(cm.output))
+        self.assertIn('handle_exceptions got error for function', str(cm.output))
+
+        with self.assertLogs() as cm:
+            resp_json = self.raise_exception(StatusCodeException(HTTPStatus.NOT_FOUND))
+            self.assertEqual(resp_json.status_code, 404)
+            self.assertEqual(resp_json.status, "404 NOT FOUND")
+            resp_msg_data = json.loads(resp_json.data.decode('utf-8'))
+            self.assertIsInstance(resp_msg_data, dict)
+            self.assertEqual(resp_msg_data['message'], 'Not Found')
+        self.assertIn('StatusCodeException', str(cm.output))
+        self.assertIn('handle_exceptions got error for function', str(cm.output))
+
+        with self.assertLogs() as cm:
+            resp_json = self.get_data()

```

Comment:
how come `get_data` raise 500?

Reply:
`get_data` returns `response_json()` which gets no parameters so that will raise `BaseException` in the decorator and that raises 500

Reply:
1. this is `implicit`. if you want to test a function that raises an exception, we need to declare it `explicitly`.
meaning a function that explicitly throws an exception need to be used.

2. I don't follow the comment. 
`response_json` calling `generate_response`  . inside this function. 
```
  if not data:
        data = b''
```
what i am missing?

Reply:
`TypeError: response_json() missing 1 required positional argument: 'data'` this is an error if we don't pass anything to it, I was testing for this, I got your point I'll rectify it